sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/TransactionCoordinator.java

```diff
@@ -101,4 +107,20 @@
                 }
             });
     }
+
+    public Mono<Void> dispose() {
+        if (isDisposed.getAndSet(true)) {
+            return Mono.empty();
+        }
+
+        if (this.sendLink != null) {
```

Comment:
```java
return Mono.fromRunnable(() -> {
    if (sendLink != null) {
        sendLink.dispose();
    }
});
```

Reply:
I would be consistent about using `this` vs not. isDisposed is a member variable but is not referenced via `this`